frontend/src/components/ReferenceInput.vue

```diff
@@ -0,0 +1,169 @@
+<script lang="ts" setup>
+import { computed, ref, watch } from "vue"
+import ComboboxInput from "@/components/ComboboxInput.vue"
+import InputField from "@/components/input/InputField.vue"
+import TextButton from "@/components/input/TextButton.vue"
+import TextInput from "@/components/input/TextInput.vue"
+import { useValidationStore } from "@/composables/useValidationStore"
+import Reference, { LegalPeriodical } from "@/domain/reference"
+import ComboboxItemService from "@/services/comboboxItemService"
+
+const props = defineProps<{
+  modelValue?: Reference
+  modelValueList?: Reference[]
+}>()
+
+const emit = defineEmits<{
+  "update:modelValue": [value: Reference]
+  addEntry: [void]
+  cancelEdit: [void]
+  removeEntry: [value?: boolean]
+}>()
+
+const reference = ref(new Reference({ ...props.modelValue }))
+const lastSavedModelValue = ref(new Reference({ ...props.modelValue }))
+
+const validationStore = useValidationStore<(typeof Reference.fields)[number]>()
+
+const legalPeriodical = computed({
+  get: () =>
+    reference?.value?.legalPeriodical
+      ? {
+          label: reference?.value?.legalPeriodical.legalPeriodicalAbbreviation,
+          value: reference?.value?.legalPeriodical,
+          additionalInformation:
+            reference?.value?.legalPeriodical.legalPeriodicalSubtitle,
+        }
+      : undefined,
+  set: (newValue) => {
+    const legalPeriodical = { ...newValue } as LegalPeriodical
```

Comment:
Ist jetzt super kleinlich, aber die Zeile kann doch eigentlich in das `if(),` oder?